### PR TITLE
 Updates to how bundleout.txt is being written cont'd.

### PR DESCRIPTION
**Pull request by acpaquette**
The bundleout.txt file from a run of jigsaw is updated.

## Description

This update includes new information (such as units and control network information) and a reformatting of older information. The code was also reworked for maintainability. 

## Related Issue

https://github.com/USGS-Astrogeology/ISIS3/issues/3227

## Motivation and Context

This results in easier to read, and more informative log files.

## How Has This Been Tested?

A full run of the entire testing suite was conducted.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by acpaquette** [comment]
@jessemapel PR for the deprecation stuff is up

File: isis/src/control/objs/BundleUtilities/BundleObservation.cpp [hunk 1]
```diff
@@ -782,60 +783,401 @@ namespace Isis {
...
-   *
-   * @internal
-   *   @history 2016-10-26 Ian Humphrey - Default values are now provided for parameters that are
-   *                           not being solved. Fixes #4464.
-   */
-  QString BundleObservation::formatBundleOutputString(bool errorPropagation, bool imageCSV) {
+ * @brief Creates and returns a formatted QString representing the bundle coefficients and
+ * parameters
+ *
+ * @depricated The function formatBundleOutputString is depricated as of ISIS 3.8.1/3.9
```
Comment (jessemapel) [inline isis/src/control/objs/BundleUtilities/BundleObservation.cpp]: This is not going into 3.8.1, so it's just as of 3.9

File: isis/src/control/objs/BundleUtilities/BundleObservation.cpp [hunk 2]
```diff
@@ -782,60 +783,401 @@ namespace Isis {
...
+ *
+ * @return @b QString Returns a formatted QString representing the BundleObservation
+ *
+ * @internal
+ *   @history 2016-10-26 Ian Humphrey - Default values are now provided for parameters that are
+ *                           not being solved. Fixes #4464.
+ */
+QString BundleObservation::formatBundleOutputString(bool errorPropagation, bool imageCSV) {
+
+  std::cerr << "The function formatBundleOutputString is depricated as of ISIS 3.8.1/3.9 and will be removed in ISIS 4.0" << std::endl;
```
Comment (jessemapel) [inline isis/src/control/objs/BundleUtilities/BundleObservation.cpp]: Same here. Split the long message over two lines in the code.

File: isis/src/control/objs/BundleUtilities/BundleObservation.h [hunk 3]
```diff
@@ -69,14 +69,27 @@ namespace Isis {
...
+   *                            the formatBundleOutputString and uses the traditional C function
+   *                            sprintf instead of QString arg chaining because it's easier to
+   *                            make the output columns align nicely.  Also, it maintains
+   *                            consistency with text output in BundleSolutionInfo.
+   *   @history 2019-06-03  Tyler Wilson - Deleted the formatBundleOutputString and added the
+   *                            functions bundleOutputCSV/bundleOutputFetchData. Combined
+   *                            with bundleOutputString these three functions will fulfill
+   *                            the same functional role formerly occuped by
+   *                            formatBundleOutputString but with reduced code duplication.
+   *
```
Comment (jessemapel) [inline isis/src/control/objs/BundleUtilities/BundleObservation.h:94]: You should add a history comment that you flagged the old function as deprecated.

**Review by jessemapel — COMMENTED** [review]
Looks good, just needs minor docs and style clean up

File: isis/src/control/objs/BundleUtilities/BundleObservation.h (hunk 3 above)
Comment (acpaquette) [reply isis/src/control/objs/BundleUtilities/BundleObservation.h:94]: :+1:

**Comment by acpaquette** [comment]
Just a heads up if this gets merged there will most likely be failing tests.

**Review by jessemapel — APPROVED** [review]

**Comment by jessemapel** [comment]
We need to get this out in 3.9 so I am going to merge it.

**Comment by jlaura** [comment]
We can absolutely get the 3.9.0_RC1 release peeled off on Monday as well. 3.8.1 should be a goal for today if the team agrees.

**Pull request merged**